### DISCUSSION TITLE
1.17?

### DISCUSSION THREAD

**Discussion by carlni2008** [Jikoo/OpenInv#24] [General]
Are u going to release a 1.17 supported version soon?

**Comment by Jikoo** [comment]
Yup, probably should be ready in a few hours.

**Reply by Jikoo** [reply]
I've managed to set aside time this morning to look at 1.17, and there're a bunch of changes. It's mostly small stuff and mostly developer QoL, so I still expect to have an initial hack out reasonably fast, but I expect to make some structural changes in the near future.

I'm shooting for 4 hours from now roughly. I still need to take care of some other work and review of overridden inventory methods takes a long time.

**Reply by Jikoo** [reply]
Here we are, 4 hours later. Granted, an hour of that was fixing whatever broke my scripts (I guess GH must have updated Bash in their Ubuntu image or something? Dunno.), but hey, 4 hours is 4 hours.

Initial build is [here](https://github.com/Jikoo/OpenInv/suites/2973498747/artifacts/67060350). I'm not confident SilentContainer will work properly - it looks like gamemode changing may require a little additional work, which is noted [here](https://github.com/Jikoo/OpenInv/blob/ff7243db6a69e9b8fc2097907b813bc76a745517/internal/v1_17_R1/src/main/java/com/lishid/openinv/internal/v1_17_R1/AnySilentContainer.java#L339-L342).

**Reply by Jikoo** [reply]
OpenInv 4.1.8 for MC 1.16.5 and 1.17.0 [has been released](https://github.com/Jikoo/OpenInv/releases/tag/4.1.8).


Time to go download the file and then manually upload to Curseforge again because my last idea for fixing the invalid YAML post-variable-replacement didn't work. Yay.
